File: keepr/Models/Keep.cs
namespace keepr.Models;

public class Generic
{
  public int Id { get; set; }
  public string CreatorId { get; set; }
  public string Name { get; set; }
  public string Description { get; set; }
  public string Img { get; set; }
  public Profile Creator { get; set; }
}

public class Keep : Generic
{
  public int Views { get; set; }
  public int Kept { get; set; }

}

public class KeepInVault : Keep
{
  public int VaultKeepId { get; set; }
}

File: keepr.client/src/models/Keep.js
export class Keep {
  constructor(data) {
    this.creator = data.creator
    this.creatorId = data.creatorId
    this.description = data.description
    this.id = data.id
    this.imgUrl = data.img
    this.keptCount = data.kept
    this.name = data.name
    this.views = data.views
    this.vaultKeepId = data.vaultKeepId || null
  }
}

// export class VaultKeep extends Keep {
//   constructor(data) {
//     this.vaultKeepId = data.vaultKeepId
//   }
// }